pkg/initiailizer_v2/model/huggingface.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from urllib.parse import urlparse
+
+import huggingface_hub
+
+import pkg.initiailizer_v2.utils.utils as utils
+
+# TODO (andreyvelich): This should be moved to SDK V2 constants.
+import sdk.python.kubeflow.storage_initializer.constants as constants
+from pkg.initiailizer_v2.model.config import HuggingFaceModelInputConfig
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%SZ",
+    level=logging.INFO,
+)
+
+
+class HuggingFace(utils.ModelProvider):
+
+    def load_config(self):
+        config_dict = utils.get_config_from_env(HuggingFaceModelInputConfig)
+        logging.info(f"Config for HuggingFace model initiailizer: {config_dict}")
+        self.config = HuggingFaceModelInputConfig(**config_dict)
+
+    def download_model(self):
+        storage_uri_parsed = urlparse(self.config.storage_uri)
+        model_uri = storage_uri_parsed.netloc + storage_uri_parsed.path
+
+        logging.info(f"Downloading model: {model_uri}")
+        logging.info("-" * 40)
+
+        if self.config.access_token:
+            huggingface_hub.login(self.config.access_token)
+
+        # TODO (andreyvelich): We should verify these patterns for different models.
+        huggingface_hub.snapshot_download(
+            repo_id=model_uri,
+            local_dir=constants.VOLUME_PATH_MODEL,
+            allow_patterns=["*.json", "*.safetensors", "*.model"],
+            ignore_patterns=["*.msgpack", "*.h5", "*.bin"],
```

Comment:
we can consider ignoring ".pt" and ".pth" files as well

Reply:
@saileshd1402 Do we know what files have .pt and .pth extension ?

Reply:
They are store model weights similar to .safetensors and .bin but are outdated now. For example: In [meta-llama/Llama-3.1-8B](https://huggingface.co/meta-llama/Llama-3.1-8B/tree/main) there is a folder called "/original" where [consolidated.00.pth](https://huggingface.co/meta-llama/Llama-3.1-8B/blob/main/original/consolidated.00.pth) stores same data as ".safetensors" but is outdated.

Reply:
@lizzzcai do you have any thoughts on whether we can exclude `.pth` and `.pt` files from model download ?
I noticed that you suggested to add the same ignore_patterns to KServe storage initializer: https://github.com/kserve/kserve/pull/3584#discussion_r1736833011

Reply:
Hi @andreyvelich, I don't see an issue to exclude `.pth`, `.pt` and `.bin`, as safetensors should be a preferred format and it has better security compared to others. Assuming that the model being downloaded provides safetensors format (most of the popular model should have).

If you want to support multiple formats, can check how [vLLM](https://github.com/vllm-project/vllm/blob/c91ed47c436f2d45299bed5eacd257e8cbc7c312/vllm/model_executor/model_loader/loader.py#L272-L284) support it.

Reply:
I see, thanks for the info!
I think, in the future we should follow the vLLM approach.